### ISSUE TITLE
use declaredClass to generate grid-id

### ISSUE THREAD

**Issue by gerpres** [SitePen/dgrid#1047]
because of the recent changes in the dgrid codebase, unfortunately we are not able to upgrade to 0.4 in one go. therefore we want to ship both 0.3 and 0.4 until all the required changes have been made.

first tests were successful - the grid can be loaded from a different package because of the relative module-identifiers - so this works well for us.

unfortunately, when instantiating a grid without a fixed id, both codebases use 'grid_'+counter to generate an id.
we would appreciate if this could be changed to the same behavior as the dojo/dijit default by using the declaredClass-property.

another problem may be the css-files - but it seems that up to now it should be fine just to use the new one

**Comment by kfranqueiro** [comment]
I don't think that `declaredClass` would really solve this, for a couple of reasons:

* dgrid doesn't set `declaredClass` in any way (it doesn't use the deprecated string parameter to `declare`, and it doesn't go out of its way to set it otherwise)
* even if it did set it, it would probably do so statically (as does dijit)

However, I think we could potentially work something in to make the automatic ID generation logic overridable (e.g. by hanging the `generateId` function off of the `List` constructor), so that you could potentially replace the logic in one version or the other.

**Comment by kfranqueiro** [comment]
I've added an `autoIdPrefix` property exposed off of the `List` constructor, which is available now on the master and dev-0.4 branches, and will be available in 0.4.1.

This will allow you to change the prefix used for auto-generated IDs in dgrid 0.4 by loading `dgrid/List` and changing `List.autoIdPrefix` before creating any instances, e.g.:

```js
require([ 'dgrid/List' ], function (List) {
    List.autoIdPrefix = 'dgrid_0.4_';
});
```

**Comment by kfranqueiro** [comment]
Almost forgot to address your other question about CSS.  Off the top of my head, I would figure there shouldn't be too much of a difference between 0.3's and 0.4's so it should be safe to just use 0.4's.

Future major releases of dgrid will be using Stylus for structural styles and skins.  If it becomes necessary to separate the styles for 0.4 and the next version, if you can isolate your "future-ready" grid instances under a specific classname, it may be feasible to build a custom stylesheet using Stylus that imports each of dgrid's structural stylesheets nested under that classname.  I can provide more info on that when the time comes if you find you need to cross that bridge.
